api/prometheus/v1/example_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package v1_test provides examples making requests to Prometheus using the
+// Golang client.
+package v1_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+)
+
+func ExampleAPI_Query() {
+	client, err := api.NewClient(api.Config{
+		Address: "http://demo.robustperception.io:9090",
```

Comment:
Just tried to open it and it is down. 
A mock client would be ok, but would make the example a bit strange.

Reply:
As I said in the other comment, I agree that a mock client is weird for an example. Since these are just being compiled and tests don't depend on them passing perhaps it is ok for robustperception to be down occasionally?

Reply:
Yeap, agree , it is a good start anyway!